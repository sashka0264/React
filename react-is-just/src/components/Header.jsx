import React from 'react';

const Header = () => {
  return (
    <header className="app-header">
        <img className="app-header__image" src="https://www.shareicon.net/data/128x128/2016/07/10/119874_apps_512x512.png"></img>
        <div>My network made on React</div>
    </header>
  );
};

export default Header;