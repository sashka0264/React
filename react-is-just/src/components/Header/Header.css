.app-header {
    grid-area: header;
    background-color: #f2f2f2;
    display: flex;
    align-items: center;
    padding: 20px;
    margin-top: 5px;
    border-radius: 2px;
    box-shadow: 0 1px 0 0 #d7d8db, 0 0 0 1px #e3e4e8;
}

.app-header__image {
    display: none;
    width: 33px;
    height: 33px;
}

.app-header__search {
    padding: 5px 10px;
}

@media (max-width: 767px) {
    .app-header__image {
        display: inline-block;
    }
}