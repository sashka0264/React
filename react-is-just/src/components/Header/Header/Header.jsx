import React from "react";
import {NavLink} from "react-router-dom";
import style from "./Header.module.css";

const Header = (props) => {
  console.log()
  return (
    <header className={style.appHeader}>
      <img className={style.appHeaderImage} src="https://siliconprice.com/img/mobileMenuIcon.png"/>

      <div className={style.appHeaderLoginBlock}>
        <NavLink to={"/login"}>{props.isAuth ? `${props.login}` : "Войти"}</NavLink>
      </div>
    </header>
  );
}

export default Header;