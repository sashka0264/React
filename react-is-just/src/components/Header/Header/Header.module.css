.appHeader {
	grid-area: header;
	display: flex;
	align-items: center;
	border-radius: 2px;
	box-shadow: 0 1px 0 0 #d7d8db, 0 0 0 1px #e3e4e8;
	background-color: rgb(74, 118, 168);
	border-bottom: 1px solid #29487d;
}

.appHeader a {
	color: white;
}


.appHeaderImage {
	display: none;
	width: 33px;
	height: 33px;
}

.appHeaderLoginBlock {
	margin-left: auto;
	padding-right: 20px;
}

@media (max-width: 767px) {
	.appHeaderImage {
		display: inline-block;
	}
}