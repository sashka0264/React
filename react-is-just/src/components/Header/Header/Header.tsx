import React from "react";
import {NavLink} from "react-router-dom";
import style from "./Header.module.css";

interface IProps {
  isAuth: boolean;
  login: string | null;
  logoutTC: any;
}

const Header = ({isAuth, login, logoutTC}: IProps) => {
  return (
    <header className={style.appHeader}>
      <img alt="menu-burger" className={style.appHeaderImage} src="https://siliconprice.com/img/mobileMenuIcon.png"/>

      <div className={style.appHeaderLoginBlock}>
        <NavLink className={style.appHeaderLogIn} to="/login">
          <span>{isAuth ? `${login}` : "Войти"}</span>
        </NavLink>
        
        <span onClick={() => logoutTC()} className={style.appHeaderLogOut}>
          {isAuth ? "Выйти" : null}
        </span>
      </div>
    </header>
  );
};

export default Header;