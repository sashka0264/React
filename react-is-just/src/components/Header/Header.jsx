import React from 'react';
import "./Header.css";

const Header = (props) => {
  return (
    <header className="app-header">
        <img className="app-header__image" src="https://siliconprice.com/img/mobileMenuIcon.png"></img>
        <input className="app-header__search" placeholder="Найти Друга"/>
    </header>
  );
};


 
export default Header;