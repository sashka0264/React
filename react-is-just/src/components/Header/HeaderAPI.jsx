import React, {Component} from 'react';
import {connect} from "react-redux";
import {getMeTC, logoutTC} from "../../redux/actions";
import Header from "./Header/Header";

class HeaderAPI extends Component {
  componentDidMount() {
    const {getMeTC} = this.props;
    getMeTC();
  }
  render() {
    return <Header {...this.props}/>
  }
};

const mapStateToProps = ({global}) => {
  return {
    isAuth: global.auth.isAuth,
    login: global.auth.login
  }
}
 
export default connect(mapStateToProps, {getMeTC, logoutTC})(HeaderAPI);