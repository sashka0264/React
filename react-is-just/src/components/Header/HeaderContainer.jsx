import React, {Component} from 'react';
import {connect} from "react-redux";
import {logoutTC} from "../../redux/actions";
import Header from "./Header/Header";

class HeaderContainer extends Component {
  render() {
    return <Header {...this.props}/>
  }
};

const mapStateToProps = ({global}) => ({isAuth: global.auth.isAuth, login: global.auth.login});
  
export default connect(mapStateToProps, {logoutTC})(HeaderContainer);