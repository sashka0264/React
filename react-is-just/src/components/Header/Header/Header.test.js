import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
import {createSerializer} from "enzyme-to-json";

expect.addSnapshotSerializer(createSerializer({mode: "deep"}));

describe("Header", () => {
    const props = {
        isAuth: true,
        login: "sasha2020",
        logoutTC: () => {}
    };

    it("The user is authorized", () => {
        const HeaderShallow = shallow(<Header {...props} />);
        expect(HeaderShallow).toMatchSnapshot();
    });

    it("The user isn't authorized", () => {
        const nextProps = {...props, isAuth: false};
        const HeaderShallow = shallow(<Header {...nextProps} />);
        expect(HeaderShallow).toMatchSnapshot();
    });
});