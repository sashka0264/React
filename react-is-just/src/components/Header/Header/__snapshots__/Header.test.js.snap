// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header The user is authorized 1`] = `
<header
  className="appHeader"
>
  <img
    alt="menu-burger"
    className="appHeaderImage"
    src="https://siliconprice.com/img/mobileMenuIcon.png"
  />
  <div
    className="appHeaderLoginBlock"
  >
    <NavLink
      className="appHeaderLogIn"
      to="/login"
    >
      <span>
        sasha2020
      </span>
    </NavLink>
    <span
      className="appHeaderLogOut"
      onClick={[Function]}
    >
      Выйти
    </span>
  </div>
</header>
`;

exports[`Header The user isn't authorized 1`] = `
<header
  className="appHeader"
>
  <img
    alt="menu-burger"
    className="appHeaderImage"
    src="https://siliconprice.com/img/mobileMenuIcon.png"
  />
  <div
    className="appHeaderLoginBlock"
  >
    <NavLink
      className="appHeaderLogIn"
      to="/login"
    >
      <span>
        Войти
      </span>
    </NavLink>
    <span
      className="appHeaderLogOut"
      onClick={[Function]}
    />
  </div>
</header>
`;
